rest_lights.cpp

```diff
@@ -1973,101 +1973,113 @@ int DeRestPluginPrivate::setWindowCoveringState(const ApiRequest &req, ApiRespon
 int DeRestPluginPrivate::setTuyaDeviceState(const ApiRequest &req, ApiResponse &rsp, TaskItem &taskRef, QVariantMap &map)
 {
     QString id = req.path[3];
+    
     bool targetOn = false;
-
-    if (map.contains("on"))
+    bool hasOn = false;
+    bool hasBri = false;
+    qint16 targetBri = 0;

```

Comment:
`uint targetBri = 0;`

`targetBri` should be `uint` otherwise the following would silently truncate out of range brightness values.

`targetBri = map["bri"].toUInt(&ok);`

Reply:
done